--- /home/syeda/vagrant/varnish-migrate3to4/files/default.vcl.v4	2016-09-27 03:29:27.432807326 +0200
+++ /home/syeda/vagrant/varnish-migrate3to4/files/default.vcl	2016-09-27 03:29:27.384806900 +0200
@@ -46,17 +46,17 @@
 	    set req.http.X-Forwarded-For = client.ip;
 	}
     }
-    if (req.method != "GET" &&
-      req.method != "HEAD" &&
-      req.method != "PUT" &&
-      req.method != "POST" &&
-      req.method != "TRACE" &&
-      req.method != "OPTIONS" &&
-      req.method != "DELETE") {
+    if (req.request != "GET" &&
+      req.request != "HEAD" &&
+      req.request != "PUT" &&
+      req.request != "POST" &&
+      req.request != "TRACE" &&
+      req.request != "OPTIONS" &&
+      req.request != "DELETE") {
         /* Non-RFC2616 or CONNECT which is weird. */
         return (pipe);
     }
-    if (req.method != "GET" && req.method != "HEAD") {
+    if (req.request != "GET" && req.request != "HEAD") {
         /* We only deal with GET and HEAD by default */
         return (pass);
     }
@@ -64,7 +64,7 @@
         /* Not cacheable by default */
         return (pass);
     }
-    return (hash);
+    return (lookup);
 }
 
 sub vcl_pipe {
@@ -78,7 +78,7 @@
 }
 
 sub vcl_pass {
-    return (fetch);
+    return (pass);
 }
 
 sub vcl_hash {
@@ -88,7 +88,7 @@
     } else {
         hash_data(server.ip);
     }
-    return (lookup);
+    return (hash);
 }
 
 sub vcl_hit {
@@ -99,7 +99,7 @@
     return (fetch);
 }
 
-sub vcl_backend_response {
+sub vcl_fetch {
     if (beresp.ttl <= 0s ||
         beresp.http.Set-Cookie ||
         beresp.http.Vary == "*") {
@@ -107,9 +107,7 @@
 		 * Mark as "Hit-For-Pass" for the next 2 minutes
 		 */
 		set beresp.ttl = 120 s;
-		# set beresp.ttl = 120s;
-		set beresp.uncacheable = true;
-		return (deliver);
+		return (hit_for_pass);
     }
     return (deliver);
 }
@@ -118,51 +116,27 @@
     return (deliver);
 }
 
-sub vcl_backend_error {
-    set beresp.http.Content-Type = "text/html; charset=utf-8";
-    set beresp.http.Retry-After = "5";
-    synthetic({"
-<?xml version="1.0" encoding="utf-8"?>
-<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
- "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
-<html>
-  <head>
-    <title>"} + beresp.status + " " + beresp.reason + {"</title>
-  </head>
-  <body>
-    <h1>Error "} + beresp.status + " " + beresp.reason + {"</h1>
-    <p>"} + beresp.reason + {"</p>
-    <h3>Guru Meditation:</h3>
-    <p>XID: "} + bereq.xid + {"</p>
-    <hr>
-    <p>Varnish cache server</p>
-  </body>
-</html>
-"});
-    return (deliver);
-}
-
-sub vcl_synth {
-    set resp.http.Content-Type = "text/html; charset=utf-8";
-    set resp.http.Retry-After = "5";
-    synthetic({"
+sub vcl_error {
+    set obj.http.Content-Type = "text/html; charset=utf-8";
+    set obj.http.Retry-After = "5";
+    synthetic {"
 <?xml version="1.0" encoding="utf-8"?>
 <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
 <html>
   <head>
-    <title>"} + resp.status + " " + resp.reason + {"</title>
+    <title>"} + obj.status + " " + obj.response + {"</title>
   </head>
   <body>
-    <h1>Error "} + resp.status + " " + resp.reason + {"</h1>
-    <p>"} + resp.reason + {"</p>
+    <h1>Error "} + obj.status + " " + obj.response + {"</h1>
+    <p>"} + obj.response + {"</p>
     <h3>Guru Meditation:</h3>
     <p>XID: "} + req.xid + {"</p>
     <hr>
     <p>Varnish cache server</p>
   </body>
 </html>
-"});
+"};
     return (deliver);
 }
 
